static/app/views/performance/newTraceDetails/trace.tsx

```diff
@@ -322,10 +321,42 @@ function Trace({
       pathname: location.pathname,
       query: queryParamsWithoutNode,
     });
+
+    setDetailNode(null);
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  useOnClickOutside(containerRef.current, onOutsideClick);
+  // Unhighlight the row when clicking outside of the trace or the detail panel
+  useEffect(() => {
+    const listener = (event: MouseEvent | TouchEvent) => {
+      if (!containerRef.current) {
+        return;
+      }
+
+      const traceViewElement: HTMLDivElement | null =
+        'current' in containerRef ? containerRef.current : containerRef;
+      const detailPanelElement: HTMLDivElement | null =
+        detailPanelRef && 'current' in detailPanelRef
+          ? detailPanelRef.current
+          : detailPanelRef;
+
+      const clickedTraceView =
+        !traceViewElement || traceViewElement.contains(event.target as Node);
+      const clickedDetailPanel = detailPanelElement?.contains(event.target as Node);
+      if (clickedTraceView || clickedDetailPanel) {
```

Comment:
cant this just be 
```suggestion
      if (traceViewElement?.contains?.(event.target) || detailElement?.contains?.(event.target) {
```
Why do we ned to care about if the panel element is there or not?

Reply:
Otherwise we lose the row highlighting, lets say during resizing. We don't want to treat the drawer as an outside click.